src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Models/DataLakeStoreFileSystemClient.cs

```diff
@@ -0,0 +1,877 @@
+                                                    cmdletRunningRequest.WriteWarning(secondEx.ToString());
+                                                }
+                                                else
+                                                {
+                                                    Console.WriteLine(secondEx);
+                                                }
+
+                                                return true;
+                                            });
+                                    }
+                                    else
+                                    {
+                                        if (cmdletRunningRequest != null)
+                                        {
+                                            cmdletRunningRequest.WriteWarning(ex.ToString());
+                                        }
+                                        else
+                                        {
+                                            Console.WriteLine(ex);
+                                        }
+                                    }
+
+                                    return true;
+                                });
+                        }
+                    }
+                }
+            }
+            finally
+            {
+                // set the max limit back to the original value.
+                ServicePointManager.DefaultConnectionLimit = previousDefaultConnectionLimit;
+                TracingAdapter.IsEnabled = originalValue;
+            }
+
+            if (allFailedDirs.Count > 0 && !cmdletCancellationToken.IsCancellationRequested)
+            {
+                var errString =
+                    "The following {0} directories could not be opened and their contents must be copied up with the single file copy command: {1}";
+                if (cmdletRunningRequest != null)
+                {
+                    cmdletRunningRequest.WriteWarning(
+                        string.Format(errString, allFailedDirs.Count, string.Join(",\r\n", allFailedDirs)));
+                }
+                else
+                {
+                    Console.WriteLine(errString, allFailedDirs.Count, string.Join(",\r\n", allFailedDirs));
+                }
+            }
+
+            if (allFailedFiles.Count > 0 && !cmdletCancellationToken.IsCancellationRequested)
+            {
+                var errString =
+                    "The following {0} files could not be copied and must be copied up with the single file copy command: {1}";
+                if (cmdletRunningRequest != null)
+                {
+                    cmdletRunningRequest.WriteWarning(
+                        string.Format(errString, allFailedFiles.Count, string.Join(",\r\n", allFailedFiles)));
+                }
+                else
+                {
+                    Console.WriteLine(errString, allFailedFiles.Count, string.Join(",\r\n", allFailedFiles));
```

Comment:
If we are debugging the folder upload without using PowerShell, we want to be able to output to the console. This client should be able to run its methods outside of a PowerShell runspace.